.circleci/config.yml

```diff
@@ -17,11 +17,20 @@ commands:
       versions:
         type: string
     steps:
+      - run:
+          name: Install CMake and native library dependencies
+          command: |
+            sudo apt-get update && sudo apt-get install -yq cmake libicu-dev
```

Comment:
There are some requirements to use sqlite native libraries, including `CMake` dependencies and `sqlite` git library. For `CMake` dependency, maybe we can `externalNativeBuild` with manual https://developer.android.com/ndk/guides/cmake, https://developer.android.com/studio/projects/install-ndk#install_ndk_and_cmake_automatically and https://developer.android.com/studio/projects/add-native-code. The SDK manager can download correctly `CMake` for host machine without adding those dependencies with `apt-get` for `Ubuntu`/`Debian`.  For `sqlite` git repository dependency, what about adding it with submodule? The developer can use submodule related git command to download all git repositories that `Robolectric` needs.

Reply:
Does the `cmake` that is bundled with the Android SDK support host builds and host toolchains? We need this native code to compile for host Linux / Mac Intel / Mac M1 / Windows. I am not sure that the Android SDK CMake supports this? From what I am reading, it is primarily to compile binaries that work in Android platforms.

Git submodule seems fine, I can look into that.

Reply:
The `CMake` is official recommended method to integrate C/C++ code to Android project. And the developer can add specific `CMake` toolchains version at Gradle. If Android Studio supports it for all platforms, I think this method can supports all platforms for C/C++ code too. We can follow https://developer.android.com/studio/projects/install-ndk#install_ndk_and_cmake_automatically to download `CMake` and its related toolchains at CI, and developers can use this method too or use SDK manager to download those tools(https://developer.android.com/studio/projects/install-ndk#default-version). `~/Android/Sdk/cmake` is default `CMake` installed directory on `Ubuntu`. If we use `apt-get` to install `CMake` and its toolchains, we also must specify version to get more compatibility for C/C++ code, that can be achieved by Gradle very easy.

Reply:
> The CMake is official recommended method to integrate C/C++ code to Android project

Yes but Robolectric itself is a Java project :) We just need to build JNI libraries for Robolectric that can run on host platforms (Windows/Mac/Linux). Are you sure that the CMake that comes with Android is designed for this use case? 

For instance, I was thinking of using similar to the conscrypt build system to build the native libraries:
https://github.com/google/conscrypt/blob/master/.github/workflows/ci.yml

They use cmake as well, though I am not sure exactly where it comes from in Windows.

Reply:
Maybe I am wrong about `CMake` to support different host machines. `conscrypt`'s building workflow is more meaningful. What about adding a building script for developers?  I often build `Robolectric` snapshot and use it locally.

Reply:
You can build it using `./gradlew :nativeruntime:makeNativeRuntime` (which runs as part of ./gradlew :nativeruntime:assemble). Though you need to clone the sqlite directory:
https://github.com/robolectric/robolectric/pull/6720/files#diff-fb64102a79365cf3b8a57fc04ede18b8f32bb7c9178d32d2f8110a8be16213c5

e.g.:
```
git clone https://android.googlesource.com/platform/external/sqlite $HOME/sqlite -b android11-release
export ANDROID_SQLITE_DIR=$HOME/sqlite
```
I'll update that tomorrow to use a git submodule so it can be done automatically and then no need to fiddle w/ environment variables.

Also, looks like the ubuntu GitHub action machine already includes CMake, so no need to install it for GitHub workflows (at least on Linux). Tomorrow I'll also just disable + delete the CircleCI build and we can standardize on GitHub actions.